src/compute-client/src/plan/mod.rs

```diff
@@ -1628,6 +1629,20 @@ This is not expected to cause incorrect results, but could indicate a performanc
             debug_name: desc.debug_name,
         };
 
+        // perform plan refinements for the dataflow
+        Plan::refine_source_mfps(&mut dataflow);
+        if enable_monotonic_oneshot_selects {
+            Plan::refine_single_time_dataflow(&mut dataflow);
+        }
+
+        mz_repr::explain::trace_plan(&dataflow);
```

Comment:
nit: if you add `instrument` calls to `refine_source_mfps` and `refine_single_time_dataflow` you can see the intermediate physical plans during finalization in the `EXPLAIN OPTIMIZER TRACE` output, which might be useful for debugging.

If you want to do that as part of this PR, please also:

1. Change the `instrument` name of the enclosing method to `finalize_dataflow`.
2. Change the `ExplainStage::PhysicalPlan`  value to `optimize/finalize_dataflow`.
3. Sed `/mir_to_lir/finalize_dataflow/` in other parts of the code.
4. Add an `instrument` to `Self::from_mir` which has the `mir_to_lir` name.

Reply:
I added instrumentation in 1b5a444508b2481a00303ac0273c46994569238d and verified locally that the output of `EXPLAIN OPTIMIZER TRACE` as well as the output on the GUI will show the new components of `optimize/finalize_dataflow`.